.github/workflows/test.yml

```
@@ -0,0 +1,57 @@
+name: "Run unit tests"
+
+on:
+  - push
+  - pull_request
+
+env:
+  COMPOSER_MEMORY_LIMIT: -1
+
+jobs:
+  test:
+    name: "Build"
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 12
+      matrix:
+        php: ['7.1', '7.2', '7.3', '7.4', '8.0', '8.1']
+        package-release: [dist]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup PHP ${{ matrix.php }}
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: exif,json,mbstring,dom
+
+      - name: Get user-level Composer cache
+        id: composer-cache
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+      - name: Setup Composer cache
+        uses: actions/cache@v1
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: composer-${{ runner.os }}-${{ matrix.php }}-${{ matrix.package-release }}-${{ hashFiles('**/composer.json') }}
+          restore-keys: |
+            composer-${{ runner.os }}-${{ matrix.php }}-${{ matrix.package-release }}-${{ env.cache-name }}-
+            composer-${{ runner.os }}-${{ matrix.php }}-${{ matrix.package-release }}-
+            composer-${{ runner.os }}-${{ matrix.php }}-
+            composer-${{ runner.os }}-
+
+      - name: Install composer dependencies
+        run: composer install --no-progress --no-interaction --prefer-${{ matrix.package-release }}
```

This behaviour is, while correct, often replaced by the [ramsey/composer-install](https://github.com/ramsey/composer-install) action. Therefore, this would be better:

```suggestion
      - name: Install composer dependencies
        uses: ramsey/composer-install@v1
        with:
          dependency-versions: ${{ matrix.package-release }}
```

But, since the `package-release` is set to `dist` and nothing else, you may just remove that last part:

```yaml
      - name: Install composer dependencies
        uses: ramsey/composer-install@v1
```

Also remove the `package-release` entry in the matrix.